tinygrad/runtime/ops_metal.py

```diff
@@ -38,27 +88,31 @@ def __init__(self, device:MetalDevice, name:str, lib:bytes):
           print("Error running disassembler: Make sure you have https://github.com/dougallj/applegpu cloned to tinygrad/extra/disassemblers/applegpu")
     assert lib[:4] == b"MTLB", "Invalid Metal library. Could be due to using conda. Try system python or METAL_XCODE=1 DISABLE_COMPILER_CACHE=1."
     data = libdispatch.dispatch_data_create(lib, len(lib), None, None)
-    self.library = unwrap2(self.device.device.newLibraryWithData_error_(data, None))
```

Comment:
all the places with that take error should probably be passed an explicit ctypes.byref(objc_instance(0)). And when it signals an error, do 
```
bytes(msg(msg(compileError, "localizedDescription", restype=objc_instance),
                                                              "UTF8String", restype=ctypes.c_char_p)).decode()
``` to convert it to something readable and then raise